plugins/services/src/js/components/ServiceStatusIcon.js

```diff
@@ -0,0 +1,172 @@
+          return (
+            DeclinedOffersUtil.displayDeclinedOffersWarning(item) ||
+            this.isUnableToLaunch(item)
+          );
+        }
+
+        return false;
+      })
+      .flattenItems()
+      .getItems().length;
+
+    if (appsWithWarningsCount > 0) {
+      return this.getTooltip(
+        `DC/OS is waiting for resources and is unable to complete the deployment of ${appsWithWarningsCount} ${StringUtil.pluralize("service", appsWithWarningsCount)} in this group.`
+      );
+    }
+
+    return null;
+  }
+
+  getTooltip(content) {
+    const { service } = this.props;
+    let icon = <Icon {...StatusIcon.WARNING} size="mini" />;
+
+    if (service instanceof Service) {
+      const servicePath = encodeURIComponent(service.getId());
+      content = getTooltipContent(service, content);
+      icon = <Link to={`/services/detail/${servicePath}/debug`}>{icon}</Link>;
+    }
+
+    return (
+      <Tooltip
+        interactive={true}
+        content={content}
+        width={250}
+        wrapText={true}
+        wrapperClassName="tooltip-wrapper"
+      >
+        {icon}
+      </Tooltip>
+    );
+  }
+
+  getUnableToLaunchWarning(service) {
+    if (this.isUnableToLaunch(service)) {
+      return this.getTooltip(
+        `DC/OS has been unable to complete this deployment for ${DateUtil.getDuration(Date.now() - DateUtil.strToMs(service.getQueue().since), null)}.`
+      );
+    }
+
+    return null;
+  }
+
+  isUnableToLaunch(service) {
+    const queue = service.getQueue();
+
+    if (queue == null) {
+      return false;
+    }
+
+    return (
+      Date.now() - DateUtil.strToMs(queue.since) >= UNABLE_TO_LAUNCH_TIMEOUT
+    );
+  }
+
+  render() {
+    const { service } = this.props;
+    const serviceStatus = service.getServiceStatus();
+    let iconState;
+
+    if (serviceStatus === ServiceStatus.RUNNING) {
+      iconState = StatusIcon.SUCCESS;
+    }
+
+    if (
+      serviceStatus === ServiceStatus.DEPLOYING ||
+      serviceStatus === ServiceStatus.RECOVERING ||
+      serviceStatus === ServiceStatus.DELETING
+    ) {
+      iconState = StatusIcon.LOADING;
+    }
+
+    if (serviceStatus === ServiceStatus.WARNING) {
+      iconState = StatusIcon.WARNING;
+    }
+
+    if (serviceStatus === ServiceStatus.STOPPED) {
+      iconState = StatusIcon.STOPPED;
+    }
+
+    if (service instanceof ServiceTree) {
+      return (
+        this.getServiceTreeWarning(service) ||
+        <Icon {...iconState} size="mini" />
+      );
+    }
+
+    // Display the declined offers warning if that's causing a delay.
+    if (DeclinedOffersUtil.displayDeclinedOffersWarning(service)) {
+      return this.getDeclinedOffersWarning(service);
+    }
+
+    // If not, we try to display an unable to launch warning, then default to nothing.
+    if (this.isUnableToLaunch(service)) {
+      return this.getUnableToLaunchWarning(service);
+    }
+
+    return <Icon {...iconState} size="mini" />;
```

Comment:
In the current cluster, sometimes the `iconState` doesn't get set and so undefined is passed into the required prop `id` in `Icon`. See screenshot below:

![screen shot 2017-11-07 at 4 28 56 pm](https://user-images.githubusercontent.com/1527504/32525380-049d878c-c3d9-11e7-8e5b-e18faf3cecb5.png)

Reply:
And this is because there is no `if` for the N/A service status. Also no `if` path for "Waiting" or "Delayed" status. Is this meant to be like this?

Reply:
@ahoskins correct Lee and I are still figuring out what to do with the N/A case.

Now for `Waiting ` and `Delayed ` that shouldn't be the case since it's now `deploying` instead.

Reply:
For the NA case I'm adding a condition to catch any other case that isn't defined before.

Reply:
@weblancaster do we need to capture that in a JIRA or have you already started the work?

Reply:
@MatApple talking to @leemunroe seems like we do need a more long-term solution for `why do we have NA? and should we keep it?` but for now, to not block this PR I have added a check for all cases that don't match the states for icons and display no icon in that case.